cura/BuildPlateModel.py

```diff
@@ -0,0 +1,82 @@
+from UM.Qt.ListModel import ListModel
+from PyQt5.QtCore import pyqtSignal, pyqtProperty, pyqtSlot
+from UM.Scene.Iterator.DepthFirstIterator import DepthFirstIterator
+from UM.Scene.SceneNode import SceneNode
+from UM.Scene.Selection import Selection
+from UM.Logger import Logger
+from UM.Application import Application
+
+
+class BuildPlateModel(ListModel):
+    maxBuildPlateChanged = pyqtSignal()
+    activeBuildPlateChanged = pyqtSignal()
+    selectionChanged = pyqtSignal()
+
+    def __init__(self):
+        super().__init__()
+        Application.getInstance().getController().getScene().sceneChanged.connect(self.updateMaxBuildPlate)  # it may be a bit inefficient when changing a lot simultaneously
+        Application.getInstance().getController().getScene().sceneChanged.connect(self._updateSelectedObjectBuildPlateNumbers)
+        Selection.selectionChanged.connect(self._updateSelectedObjectBuildPlateNumbers)
+
+        self._max_build_plate = 1  # default
+        self._active_build_plate = -1
+        self._selection_build_plates = []
+
+    @pyqtSlot(int)
+    def setActiveBuildPlate(self, nr):
```

Comment:
Why is there no manager/handler/controller to manage build plate instances instead of doing all of that in an instance of a build plate list model? I get the feeling that this list model is actually used more as a controller than a model.

Reply:
Hmm yes the model doubles as a controller, I'll split it. The only downside is that activeBuildPlateChanged then only passes the signal 1 step further.